server/state.go

```diff
@@ -0,0 +1,75 @@
+package server
+
+import (
+	"strings"
+	"time"
+)
+
+type nonceFix struct {
+	createdAt time.Time
+	txHash    string
+	numTries  uint64
+}
+
+func NewNonceFix(txHash string) *nonceFix {
+	return &nonceFix{
+		createdAt: Now(),
+		txHash:    strings.ToLower(txHash),
+	}
+}
+
+// todo: put into redis
+type GlobalState struct {
+	txForwardedToRelay  map[string]time.Time      // key: txHash
+	accountWithNonceFix map[string]*nonceFix      // key: txFrom
+	userLatestTx        map[string]StringWithTime // key: txFrom, value: txHash
+	txToUser            map[string]StringWithTime // key: txHash, value: txFrom
```

Comment:
nitpicking but `txHashToUser` and `userLatestTxHash` are more descriptive. Otherwise someone might expect the transaction object

Reply:
Good point :)